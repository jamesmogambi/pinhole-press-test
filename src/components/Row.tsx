import React from "react";
import RowItem from "./RowItem";
import Cell from "./Cell";
import { getGrandTotal } from "../selectors/categories";

interface RowProp {
  states: IState[];
  sales: ISale[];
  categories: Category[];
}

export const Row: React.FC<RowProp> = ({ states, categories, sales }) => {
  const renderGrandTotal = () => {
    return states.map((s, i) => {
      let { state } = s;
      let grandTotal = getGrandTotal(state, sales);
      return (
        <Cell
          key={i}
          styleProp="text-dark cell-dark cell-total"
          value={grandTotal}
          icon={false}
        />
      );
    });
  };

  return (
    <React.Fragment>
      <div className="container">
        {categories.map((item, i) => (
          <RowItem key={i} category={item} states={states} />
        ))}
        <div className="row-items item-border">
          <Cell
            styleProp="text-dark cell-double"
            value="Grand Total"
            icon={false}
          />
          {renderGrandTotal()}
        </div>
      </div>
    </React.Fragment>
  );
};

export default Row;
