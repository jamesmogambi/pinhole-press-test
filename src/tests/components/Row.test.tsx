import { shallow } from "enzyme";
import Row from "../../components/Row";
import sales from "../fixtures/sales";

test("should correctly render Row with props", () => {
  const categories = [{ name: "Furniture" }, { name: "Technology" }];
  const states = [
    { state: "Alabama", city: "Decatur" },
    { state: "Arizona", city: "Gilbert" },
    { state: "Arkansas", city: "Fayetteville" },
  ];

  const wrapper = shallow(
    <Row sales={sales} states={states} categories={categories} />
  );
  expect(wrapper).toMatchSnapshot();
});
