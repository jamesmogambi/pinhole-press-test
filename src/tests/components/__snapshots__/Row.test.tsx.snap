// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Row with props 1`] = `
<Fragment>
  <div
    className="container"
  >
    <Connect(RowItem)
      category={
        Object {
          "name": "Furniture",
        }
      }
      key="0"
      states={
        Array [
          Object {
            "city": "Decatur",
            "state": "Alabama",
          },
          Object {
            "city": "Gilbert",
            "state": "Arizona",
          },
          Object {
            "city": "Fayetteville",
            "state": "Arkansas",
          },
        ]
      }
    />
    <Connect(RowItem)
      category={
        Object {
          "name": "Technology",
        }
      }
      key="1"
      states={
        Array [
          Object {
            "city": "Decatur",
            "state": "Alabama",
          },
          Object {
            "city": "Gilbert",
            "state": "Arizona",
          },
          Object {
            "city": "Fayetteville",
            "state": "Arkansas",
          },
        ]
      }
    />
    <div
      className="row-items item-border"
    >
      <Cell
        icon={false}
        styleProp="text-dark cell-double"
        value="Grand Total"
      />
      <Cell
        icon={false}
        key="0"
        styleProp="text-dark cell-dark cell-total"
        value={1694}
      />
      <Cell
        icon={false}
        key="1"
        styleProp="text-dark cell-dark cell-total"
        value={571}
      />
      <Cell
        icon={false}
        key="2"
        styleProp="text-dark cell-dark cell-total"
        value={0}
      />
    </div>
  </div>
</Fragment>
`;
